### DISCUSSION TITLE
How to modify rewards configuration?

### DISCUSSION THREAD

**Discussion by apsarode** [trackmania-rl/tmrl#86] [Q&A]
I am trying to implement some of Stable-baselines-3 models on tmrl and wanted to get some advice from others who have used tmrl. 

How are others modifying how rewards are returned? I've struggled to find much documentation on how reward is returned from tmrl. I know that `config.json` contains some reward configurations, but could use some help what some of the parameters mean (i.e how often does the `CONSTANT_PENALTY` get returned, when is `CHECK_FORWARD` being returned, etc.)

**Comment by yannbouteiller** [comment]
We should definitely document the parameters in `config.json` better.

You can find how the reward is computed [here](https://github.com/trackmania-rl/tmrl/blob/master/tmrl%2Fcustom%2Futils%2Fcompute_reward.py).

CONSTANT_PENALTY is present at each step, but it defaults to 0 as a similar effect is achieved by gamma-discounting.

CHECK_FORWARD is the distance (number of forward "reward checkpoints") used to  check whether the car has taken a shortcut, the higher this value, the longer it is possible to cut, at the cost of higher computation required to compute the reward.

MAX_STRAY is the maximum distance the car is allowed to stray away from "reward checkpoints", after which the episode gets terminated (because the car strayed too far away from the track).
